### PR TITLE
Update seastar submodule

### PR DESCRIPTION
**Pull request by avikivity**
Drop local formatter for seastar::http::reply, which should have
been added to Seastar in the first place, and now conflicts. Also
drop local formatters for types that are aliases for Seastar types
which have gained formatters.

Disable recently-gained TLS use of OpenSSL instead of gnutls. We
don't need it, and it causes link errors with LTO.

Fix incorrect skipping in encrypted_file_test, which computed
the remaining stream length but did not account for already
consumed size_to_compare.

Change utils::gcp::storage::client::object_data_source::skip()
to match new Seastar behavior (rejecting skip-past-eof with an
exception). This is needed since https://github.com/scylladb/scylladb/commit/30f1075544272a79c05bf29234a2a838e8e946f3 switched the test's
data source to a Seastar implementation. It is also more correct -
if we're asked to skip n bytes but the stream doesn't have n bytes,
this is a protocol violation.

Contains test fix from Pavel, exposed by [1]:

test: Handle premature EOF in test_gcp_storage_skip_read

The test intentionally uses file_size larger than the actual object to
exercise EOF behavior. When input_stream::skip() is called after EOF,
it throws std::runtime_error("premature end of stream"). Catch this
specific exception from both streams, verify they agree, and exit the
loop gracefully.

Signed-off-by: Pavel Emelyanov <xemul@scylladb.com>

Co-authored-by: Copilot <223556219+Copilot@users.noreply.github.com>

[1] https://github.com/scylladb/seastar/commit/cbd1e17d2f16151ca304db533a2a65c39a7f4631, included in this Seastar submodule update

* seastar 4d268e0e...485a62b2 (50):
  > reactor: open_directory(): honor bypass_fsync
  > http: Add formatters for http::request and http::reply
  > Merge 'Assorted set of io-tester cleanups' from Pavel Emelyanov
    io_tester: Remove unused and internal-only accessor
    io_tester: Move think-time machinery into thinker_state
    io_tester: Move _file to io_class_data
    io_tester: Replace class_data::_start member with a local variable
    io_tester: Move _alignment from class_data to io_class_data
    io_tester: Remove buffer allocation from top-level request issuing
    io_tester: Cleanup context::stop() invocation
    io_tester: Allocate write buffer once to fill a file
    io_tester: Declare quantiles arrays as static constexpr
    io_tester: Drop class_data::type_str()
    io_tester: Replace != "" comparisons with .empty()
    io_tester: Replace gen_class_data() if/else chain with a switch
    io_tester: Deduplicate vectorized I/O classes
  > io_tester: fix crash from missing metric during startup
  > net: tls: adjust openssl integration to new module support
  > http/client: Count and export integrated queue length
  > Merge 'Introduce pipe_data_source_impl and pipe_data_sink_impl' from Pavel Emelyanov
    fstream: add pipe_data_source_impl and pipe_data_sink_impl
    pollable_fd: add write_some/write_all backed by writev
    pollable_fd: rename write_some/write_all(iovec) to send_some/send_all
  > reactor: Make pollable_fd_state helper methods private
  > module: extend seastar.cppm with comprehensive public API exports
  > Merge 'Add exhaustive input_stream invariant test + fixes' from Pavel Emelyanov
    tests: add exhaustive input_stream read/skip invariant test
    iostream: make skip() reject premature end of stream with exception
  > Merge 'Allow runtime selectability of GnuTLS or OpenSSL' from Noah Watkins
    net/tls: avoid potential read-past-buffer
    net/tls: move credential methods to generic tls layer
    net/tls: rename credentials_impl::dh_params to set_dh_params
    test/tls: enable openssl tls unit test
     test/tls: fix CA cert generation to use v3_ca extensions
    github: disable parallel test execution in alpine workflow
    crypto: support compiling seastar without gnutls
    net/tcp: use crypto provider for md5 calculation
    tls: fix test_peer_certificate_chain_handling for OpenSSL
    net/tls: fix test for self-signed server cert opoenssl compat
    net/tls: disable priority strings test for openssl provider
    core/crypto: expose crypto backend name for introspection
    test/tls: remove gnutls version guard
    net/tls: add openssl tls backend
    http: use backend agnostic tls error code
    net/tls: make error codes configurable by each tls backend
    net/tls: move reloadable_credentials to generic tls layer
    net/tls: move build_certificate to generic tls layer
    net/tls: move apply_to() to generic tls layer
    net/tls: move credential methods to generic tls layer
    net/tls: add OpenSSL-specific methods to public API with no-op defaults
    net/tls: introduce dh_params and credentials abstraction layer
    net/tls: add credentials_impl abstract base class
    net/tls: dispatch tls::error_category() through crypto_provider
    net/tls: dispatch wrap_client/wrap_server through crypto_provider
    net/tls: add tls_backend interface to crypto_provider
    net/tls: move public tls API methods to generic tls layer
    net/tls: move formatting utilities to generic tls layer
    net/tls: move credentials_builder blob methods to generic tls layer
    net/tls: move dh_params::from_file to generic tls layer
    net/tls: move abstract_credentials file methods to generic tls layer
    net/tls: move tls_socket_impl to generic tls layer
    net/tls: move server_session to general tls layer
    net/tls: move tls_connected_socket_impl to generic tls layer
    net/tls: move net::get_impl to generic tls layer
    net/tls: move session_ref to generic tls layer
    net/tls: add session_impl abstract interface for tls pluggability
    net/tls: rename tls.cc to be gnutls specific
    crypto: introduce crypto provider abstraction
    http: remove unused include
  > tls: test_send_two_large
  > rpc: include exception type for remote errors
  > GHA: increase timeout to 60 minutes
  > apps/httpd: replace deprecated reply::done() with write_body()
  > missing header(s)
  > net: Fix missing throw for runtime_error in create_native_net_device
  > tests/io_queue: account for token bucket refill granularity in bandwidth checks
  > Merge 'iovec: fix iovec_trim_front infinite loop on zero-length iovecs' from Travis Downs
    tests: add regression tests for zero-length iovec handling
    iovec: fix iovec_trim_front infinite loop on zero-length iovecs
  > util/process: graduate process management API from experimental
  > cooking: don't register ready.txt as a build output
  > sstring: make make_sstring not static
  > Add SparkyLinux to debian list in install-dependencies.sh
  > http: allow control over default response headers
  > Merge 'chunked_fifo: make cached chunk retention configurable' from Brandon Allard
    tests/perf: add chunked_fifo microbenchmarks
    chunked_fifo: set the default free chunk retention to 0
    chunked_fifo: make free chunk retention configurable
  > Merge 'reactor_backend: fix pollable_fd_state_completion reuse in io_uring' from Kefu Chai
    tests: add regression test for pollable_fd_state_completion reuse
    reactor_backend: use reset() in AIO and epoll poll paths
    reactor_backend: fix pollable_fd_state_completion reuse after co_await in io_uring
  > Merge 'coroutine: Generator cleanups' from Kefu Chai
    coroutine/generator: extract schedule_or_resume helper
    coroutine/generator: remove unused next_awaiter classes
    coroutine/generator: remove write-only _started field
    coroutine/generator: assert on unreachable path in buffered await_resume
    coroutine/generator: add elements_of tag and #include <ranges>
    coroutine/generator: add empty() to bounded_container concept
  > cmake: bump minimum Boost version to 1.79.0
  > seastar_test: remove unnecessary headers
  > cmake: bump minimum GnuTLS version to 3.7.4
  > Merge 'reactor: add get_all_io_queues() method' from Travis Downs
    tests: add unit test for reactor::get_all_io_queues()
    reactor: add get_all_io_queues() method
    reactor: move get_io_queue and try_get_io_queue to .cc file
  > http: deprecate reply::done(), remove _response_line dead field
  > core: Deprecate scattered_message
  > ci: add workflow dispatch to tests workflow
  > perf_tests: exit non-zero when -t pattern matches no tests
  > Replace duplicate SEGV_MAPERR check in sigsegv_action() with SEGV_ACCERR.
  > perf_tests: add total runtime to json output
  > Merge 'Relax large allocation error originating from json_list_template' from Robert Bindar
    implement move assignment operator for json_list_template
    json_list_template copy assignment operator reserves capacity upfront
  > perf_tests: add --no-perf-counters option
  > Merge 'Fix to_human_readable_value() ability to work with large values' from Pavel Emelyanov
    memory: Add compile-time test for value-to-human-readable conversion
    memory: Extend list of suffixes to have peta-s
    memory: Fix off-by-one in suffix calculation
    memory: Mark to_human_readable_value() and others constexpr
  > http: Improve writing of response_line() into the output
  > Merge 'websocket: add template parameter for text/binary frame mode and implement client-side WebSocket' from wangyuwei
    websocket: add template parameter for text/binary frame mode
    websocket: impl client side websocket function
  > file: Fix checks for file being read-only
  > reactor: Make do_dump_task_queue a task_queue method
  > Merge 'Implement fully mixed mode for output_stream-s' from Pavel Emelyanov
    tests/output_stream: sample type patterns in sanitizer builds
    tests/output_stream: extend invariant test to cover mixed write modes
    iostream: allow unrestricted mixing of buffered and zero-copy writes
    tests/output_stream: remove obsolete ad-hoc splitting tests
    tests/output_stream: add invariant-based splitting tests
    iostream: rename output_stream::_size to ::_buffer_size
  > reactor_backend: replace virtual bool methods with const bool_class members
  > resource: Avoid copying CPU vector to break it into groups
  > perf_tests: increase overhead column precision to 3 decimal places
  > Merge 'Move reactor::fdatasync() into posix_file_impl' from Pavel Emelyanov
    reactor: Deprecate fdatasync() method
    file: Do fdatasync() right in the posix_file_impl::flush()
    file: Propagate aio_fdatasync to posix_file_impl
    reactor: Move reactor::fdatasync() code to file.cc
    reactor,file: Make full use of file_open_options::durable bit
    file: Add file_open_options::durable boolean
    file: Account io_stats::fsyncs in posix_file_impl::flush()
    reactor: Move _fsyncs counter onto io_stats
  > http: Remove connection::write_body()

### CODE REVIEW

**Comment by avikivity** [comment]
@scylladb/releng what's this null failure?

**Comment by avikivity** [comment]
v2: drop local formatter for seastar::http::reply

**Comment by avikivity** [comment]
v3: drop more local formatters, which were referencing seastar formatters via typedefs.

**Comment by avikivity** [comment]
v4: disable TLS-via-OpenSSL, which we don't use, and which causes link errors with LTO.

**Comment by xemul** [comment]
> [test_encrypted_data_source_fuzzy.dev.1](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28637/artifact/testlog/x86_64/pytest_tests_logs/boost-encrypted_file_test.cc-test_encrypted_data_source_fuzzy.dev.1-call-90bad.log)

`unknown location(0): fatal error: in "test_encrypted_data_source_fuzzy": std::runtime_error: premature end of stream`

This is https://github.com/scylladb/seastar/pull/3308: encrypted data source seek()-ed behind end-of-stream and got new exception

**Comment by xemul** [comment]
Presumably, [gcs_tests/test_gcp_storage_skip_read.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/28637/testReport/junit/boost.gcp_object_storage_test/cc/gcs_tests_test_gcp_storage_skip_read_dev_1) is the same, but I cannot find it in logs

**Comment by xemul** [comment]
```patch
--- a/test/boost/encrypted_file_test.cc
+++ b/test/boost/encrypted_file_test.cc
@@ -579,7 +579,13 @@ static future<> test_random_data_source(std::vector<size_t> sizes) {
             auto v2 = std::string_view(unified_buff.get() + pos, size_to_compare);
             BOOST_REQUIRE_EQUAL(v1, v2);
             auto skip = unified_buff.size() - pos > 4113 ? 4097 : (unified_buff.size() - pos)/2;
-            co_await encrypted_source.skip(skip);
+            try {
+                co_await encrypted_source.skip(skip);
+            } catch (const std::runtime_error& e) {
+                // skip() may go past EOF, handle it as normal end-of-stream.
+                BOOST_REQUIRE_EQUAL(std::string_view(e.what()), "premature end of stream");
+                break;
+            }
             pos += size_to_compare + skip;
         }
         co_await encrypted_source.close();
```

@avikivity , this fixes the test_encrypted_data_source_fuzzy test

**Comment by avikivity** [comment]
> ```diff
> --- a/test/boost/encrypted_file_test.cc
> +++ b/test/boost/encrypted_file_test.cc
> @@ -579,7 +579,13 @@ static future<> test_random_data_source(std::vector<size_t> sizes) {
>              auto v2 = std::string_view(unified_buff.get() + pos, size_to_compare);
>              BOOST_REQUIRE_EQUAL(v1, v2);
>              auto skip = unified_buff.size() - pos > 4113 ? 4097 : (unified_buff.size() - pos)/2;
> -            co_await encrypted_source.skip(skip);
> +            try {
> +                co_await encrypted_source.skip(skip);
> +            } catch (const std::runtime_error& e) {
> +                // skip() may go past EOF, handle it as normal end-of-stream.
> +                BOOST_REQUIRE_EQUAL(std::string_view(e.what()), "premature end of stream");
> +                break;
> +            }
>              pos += size_to_compare + skip;
>          }
>          co_await encrypted_source.close();
> ```
> 
> @avikivity , this fixes the test_encrypted_data_source_fuzzy test

Please post it as a separate patch request with an explanation.

I claimed before that skip(n) where there are fewer than n remaining bytes in the stream is an error, but really, the remote is allowed to prematurely close.

skip() should really return either "was able to skip" or "hit eof before the skip". An exception isn't the best way to communicate that. Perhaps we should return the number of bytes skipped, like read_exactly().

**Comment by avikivity** [comment]
I guess, if it's only a test, it's not as bad an indication that the change was wrong.

**Comment by xemul** [comment]
> > @avikivity , this fixes the test_encrypted_data_source_fuzzy test
> 
> Please post it as a separate patch request with an explanation.

https://github.com/avikivity/scylladb/pull/14 ?

> I claimed before that skip(n) where there are fewer than n remaining bytes in the stream is an error, but really, the remote is allowed to prematurely close.
> 
> skip() should really return either "was able to skip" or "hit eof before the skip". An exception isn't the best way to communicate that. Perhaps we should return the number of bytes skipped, like read_exactly().

Yes, we [discussed it](https://github.com/scylladb/seastar/pull/3308#discussion_r2977409106), either option was breaking :(

**Comment by xemul** [comment]
> Presumably, [gcs_tests/test_gcp_storage_skip_read.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/28637/testReport/junit/boost.gcp_object_storage_test/cc/gcs_tests_test_gcp_storage_skip_read_dev_1) is the same, but I cannot find it in logs

Yup, it's the same.  The test creates a file of 12MiB object, but size to process is 12MiB + 2719. So when pos gets close to (but still below) object size, skip throws

**Comment by avikivity** [comment]
> > > @avikivity , this fixes the test_encrypted_data_source_fuzzy test
> > 
> > 
> > Please post it as a separate patch request with an explanation.
> 
> [avikivity#14](https://github.com/avikivity/scylladb/pull/14) ?

Why against my fork?

Won't it work against current master?

> 
> > I claimed before that skip(n) where there are fewer than n remaining bytes in the stream is an error, but really, the remote is allowed to prematurely close.
> > skip() should really return either "was able to skip" or "hit eof before the skip". An exception isn't the best way to communicate that. Perhaps we should return the number of bytes skipped, like read_exactly().
> 
> Yes, we [discussed it](https://github.com/scylladb/seastar/pull/3308#discussion_r2977409106), either option was breaking :(

Well, I thought throwing on skip eof is breaking in theory but not in practice. Looks like I was wrong.

**Comment by xemul** [comment]
> > > > @avikivity , this fixes the test_encrypted_data_source_fuzzy test
> > > 
> > > 
> > > Please post it as a separate patch request with an explanation.
> > 
> > 
> > [avikivity#14](https://github.com/avikivity/scylladb/pull/14) ?
> 
> Why against my fork?

Because it's a fix for your fork

> Won't it work against current master?

It will be a no-op against current master, as in it skip() doesn't throw 'prepature end of stream'

**Comment by xemul** [comment]
> Yup, it's the same. The test creates a file of 12MiB object, but size to process is 12MiB + 2719. So when pos gets close to (but still below) object size, skip throws

I sort of have a fix (from Claude), but it cannot run the test, it hangs at fixture creation (fake gcs server with bucket) :man_facepalming:

**Comment by xemul** [comment]
> > Yup, it's the same. The test creates a file of 12MiB object, but size to process is 12MiB + 2719. So when pos gets close to (but still below) object size, skip throws
> 
> I sort of have a fix (from Claude), but it cannot run the test, it hangs at fixture creation (fake gcs server with bucket) 🤦‍♂️

https://github.com/avikivity/scylladb/pull/15

**Comment by xemul** [comment]
> Well, I thought throwing on skip eof is breaking in theory but not in practice. Looks like I was wrong.

These are two tests that are deliberately "looking for trouble". Both skips-past-EOF are on hand-crafted memory sinks

**Comment by avikivity** [comment]
Ok. But what happens in production code if we have the same behavior?

Arguably it behaves better. Before the skip would silently ignore the missing bytes, now it throws, so we will react more correctly.

In practice there's probably no difference. As soon as we read everything after the skip, we will see eof.

**Comment by xemul** [comment]
In practice EOF in skipping should mean protocol error. Sender is not supposed to know whether receiver is going to skip this message (or its part) or not, so it should send all bytes subject for reading and maybe closing the connection after it. Doing it earlier is some exceptional case

**Comment by wmitros** [comment]
We probably want to include [00867a5](https://github.com/scylladb/seastar/commit/00867a5c8a2929c9833538efa68b9dd48d1aadd4) in the next seastar update as it fixes the CI stability issue SCYLLADB-1798

**Comment by avikivity** [comment]
> > > Yup, it's the same. The test creates a file of 12MiB object, but size to process is 12MiB + 2719. So when pos gets close to (but still below) object size, skip throws
> > 
> > 
> > I sort of have a fix (from Claude), but it cannot run the test, it hangs at fixture creation (fake gcs server with bucket) 🤦‍♂️
> 
> [avikivity#15](https://github.com/avikivity/scylladb/pull/15)

Ah, I see it doesn't work before the Seastar update.

**Comment by avikivity** [comment]
v5:
 - rrrrrrrrrrrebased
 - folded in test fix from @xemul 
 - fixed encrypted_file_test overeager skip()ping which triggered tighter check in seastar

**Comment by avikivity** [comment]
@scylladb/scylla-maint please review. Even though it's a seastar submodule update, it contains new content too.

**Review by denesb — APPROVED** [review]

**Comment by avikivity** [comment]
Dequeued. Stealth GCP tests were not executed (https://scylladb.atlassian.net/browse/SCYLLADB-1945) so this broke next.

**Comment by wmitros** [comment]
@avikivity is it possible to update all the way to [00867a5](https://github.com/scylladb/seastar/commit/00867a5c8a2929c9833538efa68b9dd48d1aadd4)? This fixes CI issue [SCYLLADB-1798](https://scylladb.atlassian.net/browse/SCYLLADB-1798) (and the issue is potentially fatal and not limited to CI)

[SCYLLADB-1798]: https://scylladb.atlassian.net/browse/SCYLLADB-1798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by avikivity** [comment]
> @avikivity is it possible to update all the way to [00867a5](https://github.com/scylladb/seastar/commit/00867a5c8a2929c9833538efa68b9dd48d1aadd4)? This fixes CI issue [SCYLLADB-1798](https://scylladb.atlassian.net/browse/SCYLLADB-1798) (and the issue is potentially fatal and not limited to CI)

Better two large bites than one very large bite.

[SCYLLADB-1798]: https://scylladb.atlassian.net/browse/SCYLLADB-1798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by avikivity** [comment]
I'm even more confused now, I managed to run it locally against real GCP, and it passes.

**Comment by avikivity** [comment]
To add to the confusion, GS_BUCKET_FOR_TEST is not defined in scylla-pkg.

**Comment by avikivity** [comment]
Rebasing to master reproduced the problem.

Culprit is 30f1075544272a79c05bf29234a2a838e8e946f3.

**Comment by avikivity** [comment]
v6:
 - rebased past https://github.com/scylladb/scylladb/commit/30f1075544272a79c05bf29234a2a838e8e946f3
 - added fix for gcp object storage data source skip() to throw if skipping past eof, matching new Seastar behavior

**Comment by xemul** [comment]
```
INFO  2026-05-12 20:48:46,926 seastar - Reactor backend: linux-aio
INFO  2026-05-12 20:48:46,927 seastar - Perf-based stall detector creation failed (EACCESS), try setting /proc/sys/kernel/perf_event_paranoid to 1 or less to enable kernel backtraces: falling back to posix timer.
INFO  2026-05-12 20:48:46,927 [shard 0:main] io - IO queue was unable to find a suitable maximum request length, the search was cut-off early at: 16MB
INFO  2026-05-12 20:48:46,927 [shard 0:main] io - IO queue was unable to find a suitable maximum request length, the search was cut-off early at: 16MB
random-seed=2563415480
Error: rootlessport listen tcp 0.0.0.0:36693: bind: address already in use
unknown location(0): fatal error: in "gcs_tests": std::bad_alloc: std::bad_alloc
```

it's _not_ https://github.com/scylladb/seastar/issues/3169

**Comment by avikivity** [comment]
> ```
> INFO  2026-05-12 20:48:46,926 seastar - Reactor backend: linux-aio
> INFO  2026-05-12 20:48:46,927 seastar - Perf-based stall detector creation failed (EACCESS), try setting /proc/sys/kernel/perf_event_paranoid to 1 or less to enable kernel backtraces: falling back to posix timer.
> INFO  2026-05-12 20:48:46,927 [shard 0:main] io - IO queue was unable to find a suitable maximum request length, the search was cut-off early at: 16MB
> INFO  2026-05-12 20:48:46,927 [shard 0:main] io - IO queue was unable to find a suitable maximum request length, the search was cut-off early at: 16MB
> random-seed=2563415480
> Error: rootlessport listen tcp 0.0.0.0:36693: bind: address already in use
> unknown location(0): fatal error: in "gcs_tests": std::bad_alloc: std::bad_alloc
> ```
> 
> it's _not_ [scylladb/seastar#3169](https://github.com/scylladb/seastar/issues/3169)

There's so little information there.

Could the EADDRINUSE be related? Something fails, then the test goes crazy?

I don't think so, there's probably a retry and the test runs, just bad_allocs for some reason.

**Comment by avikivity** [comment]
--repeat 1000, nothing

**Comment by avikivity** [comment]
--repeat 10000 failed once:

```
Running 1 test case...
INFO  2026-05-13 14:20:56,544 seastar - Reactor backend: linux-aio
INFO  2026-05-13 14:20:56,545 [shard 0:main] io - IO queue was unable to find a suitable maximum request length, the search was cut-off early at: 16MB
INFO  2026-05-13 14:20:56,545 [shard 0:main] io - IO queue was unable to find a suitable maximum request length, the search was cut-off early at: 16MB
random-seed=61260632
Error: pasta failed with exit code 1:
Listen failed for HOST TCP port */43655: Address already in use
Couldn't listen on requested TCP ports

unknown location(0): fatal error: in "gcs_tests": seastar::broken_promise: broken promise

*** 1 failure is detected in the test module "Master Test Suite"
```

of note:
 - seen EADDRINUSE
 - different failure (broken_promise vs bad_alloc)

**Comment by avikivity** [comment]
Repreoduces broken_promise a few more times, p=0.1%, always with EADDRINUSE.

**Comment by avikivity** [comment]
Filed https://scylladb.atlassian.net/browse/SCYLLADB-2003

@scylladbbot trigger-ci
